homeassistant/components/sia/__init__.py

```diff
@@ -0,0 +1,269 @@
+"""The sia integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pysiaalarm.aio import SIAAccount, SIAClient, SIAEvent
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_MOISTURE,
+    DEVICE_CLASS_SMOKE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_PORT,
+    CONF_SENSORS,
+    CONF_ZONE,
+    DEVICE_CLASS_TIMESTAMP,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.util.dt import utcnow
+from homeassistant.util.json import load_json
+
+from .alarm_control_panel import SIAAlarmControlPanel
+from .binary_sensor import SIABinarySensor
+from .const import (
+    CONF_ACCOUNT,
+    CONF_ACCOUNTS,
+    CONF_ENCRYPTION_KEY,
+    CONF_PING_INTERVAL,
+    CONF_ZONES,
+    DEVICE_CLASS_ALARM,
+    DOMAIN,
+    HUB_SENSOR_NAME,
+    HUB_ZONE,
+    LAST_MESSAGE,
+    PLATFORMS,
+    UTCNOW,
+)
+from .sensor import SIASensor
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the sia component."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up sia from a config entry."""
+    hass.data[DOMAIN][entry.entry_id] = SIAHub(
+        hass, entry.data, entry.entry_id, entry.title
+    )
+    await hass.data[DOMAIN][entry.entry_id].async_setup_hub()
```

Comment:
I think it'll be easier to follow if you have a local pointing at the hub instead of reaching into data every time you use it within the function